src/parser.rs

```diff
@@ -2219,19 +2219,20 @@ impl<'input, 'abbrev, 'unit, Endian> EntriesCursor<'input, 'abbrev, 'unit, Endia
     ///          first_entry_with_no_children.unwrap());
     /// ```
     pub fn next_dfs(&mut self) -> Option<isize> {
-        match self.current() {
-            Some(Ok(current)) => {
-                self.input = if let Some(after_attrs) = current.after_attrs.get() {
+        self.current_ref();
+        {
+            if self.cached_current.is_some() {
+                self.input = if let Some(after_attrs) = self.cached_current.as_ref().unwrap().after_attrs.get() {
                     after_attrs
                 } else {
-                    for _ in current.attrs() {
+                    for _ in self.cached_current.as_ref().unwrap().attrs() {
                     }
-                    current.after_attrs
+                    self.cached_current.as_ref().unwrap().after_attrs
                         .get()
                         .expect("should have after_attrs after iterating attrs")
                 };
 
-                let mut delta_depth = if current.abbrev.has_children() {
+                let mut delta_depth = if self.cached_current.as_ref().unwrap().abbrev.has_children() {
```

Comment:
Agreed that we need to do something... I didn't spend much time on making it clean because it still needs other changes. Same for your other comment. I hadn't even run rustfmt on this. I'll push a commit for rustfmt so that it doesn't conflict too much with your branch.